src/sentry/static/sentry/app/components/events/interfaces/debugMeta-v2/debugImageDetails/candidates.tsx

```diff
@@ -22,69 +34,379 @@ type Props = {
   isLoading: boolean;
 };
 
-function Candidates({
-  candidates,
-  organization,
-  projectId,
-  baseUrl,
-  builtinSymbolSources,
-  onDelete,
-  isLoading,
-}: Props) {
-  return (
-    <Wrapper>
-      <Title>
-        {t('Debug Files')}
-        <QuestionTooltip
-          title={tct(
-            'These are the Debug Information Files (DIFs) corresponding to this image which have been looked up on [docLink:symbol servers] during the processing of the stacktrace.',
-            {
-              docLink: (
-                <ExternalLink href="https://docs.sentry.io/platforms/native/data-management/debug-files/#symbol-servers" />
-              ),
-            }
+type State = {
+  searchTerm: string;
+  filterOptions: FilterOptions;
+  filteredCandidatesBySearch: Image['candidates'];
+  filteredCandidatesByFilter: Image['candidates'];
+};
+
+class Candidates extends React.Component<Props, State> {
+  state: State = {
+    searchTerm: '',
+    filterOptions: {},
+    filteredCandidatesBySearch: [],
+    filteredCandidatesByFilter: [],
+  };
+
+  componentDidMount() {
+    this.getFilters();
+  }
+
+  componentDidUpdate(prevProps: Props, prevState: State) {
+    if (!isEqual(prevProps.candidates, this.props.candidates)) {
+      this.getFilters();
+      return;
+    }
+
+    if (prevState.searchTerm !== this.state.searchTerm) {
+      this.doSearch();
+    }
+  }
+
+  filterCandidatesBySearch() {
+    const {searchTerm, filterOptions} = this.state;
+    const {candidates} = this.props;
+
+    if (!searchTerm.trim()) {
+      const filteredCandidatesByFilter = this.getFilteredCandidatedByFilter(
+        candidates,
+        filterOptions
+      );
+
+      this.setState({
+        filteredCandidatesBySearch: candidates,
+        filteredCandidatesByFilter,
+      });
+      return;
+    }
+
+    // Slightly hacky, but it works
+    // the string is being `stringfy`d here in order to match exactly the same `stringfy`d string of the loop
+    const searchFor = JSON.stringify(searchTerm)
```

Comment:
Do we have enough common behavior between this and Breadcrumbs that we could have a generalized component for this dropdown + search pattern?

Reply:
yes I copied it from the breadcrumbs :D and thought also about creating a common component. I will do that in a follow-up PR 😉